source/common/http/conn_manager_impl.h

```diff
@@ -253,6 +253,11 @@ class ConnectionManagerConfig {
    * @return tracing config.
    */
   virtual const TracingConnectionManagerConfig* tracingConfig() PURE;
+
+  /**
+   * @return ConnectionManagerStats& the listener stats to write to.
+   */
+  virtual ConnectionManagerStats& listenerStats() PURE;
```

Comment:
Unless you are going to increment all stats in the context of the listener (which I don't think we want to do), I don't think it makes sense to create the entire tree for the listener. I would just do 2xx/3xx/4xx/5xx and make sure it is under the http.{name} namespace inside the listener scope.

Reply:
I was debating about this technique. Not sure if eventually we wanted to get all of the stats for ConnectionManager. I'll scope this to 2xx/3xx/4xx/5xx and ping back when the change is in.